lib/collective.lib.js

```diff
@@ -150,3 +151,12 @@ export const getContributeRoute = collective => {
 export const getSuggestedTags = collective => {
   return collective?.expensesTags?.map(({ tag }) => tag) || [];
 };
+
+/** Checks if recurring contributions are allowed for the user for a given collective **/
+export const canContributeRecurring = (collective, user) => {
+  return (
+    collective.host.supportedPaymentMethods.some(paymentMethod =>
+      ProvidersWithRecurringPaymentSupport.includes(paymentMethod),
+    ) || Boolean(user?.memberOf.some(member => member.collective?.host?.id === collective.host.legacyId))
```

Comment:
This one should not be necessary, `some` already returns a boolean

```suggestion
    ) || user?.memberOf.some(member => member.collective?.host?.id === collective.host.legacyId)
```

Reply:
Done. 👍🏽